containers/ddev-dbserver/files/create_base_db.sh

```diff
@@ -23,23 +25,24 @@ mysqld_version=${mysqld_version%%-*}
 mysqld_version=${mysqld_version%.*}
 echo version=$mysqld_version
 # Oracle mysql 5.7+ deprecates mysql_install_db
-if [ "${mysqld_version}" = "5.7" ] || [  "${mysqld_version%%%.*}" = "8.0" ]; then
-    mysqld --defaults-file=/var/tmp/my.cnf --initialize-insecure --datadir=/var/lib/mysql --server-id=0
+if [ "${mysqld_version}" = "5.7" ] || [[  "${mysqld_version%%%.*}" =~ ^8.[04]$ ]]; then
+    mysqld --defaults-file=/var/tmp/my.cnf --initialize-insecure --datadir=${DATADIR:-/var/lib/mysql} --server-id=0
 else
     # mysql 5.5 requires running mysql_install_db in /usr/local/mysql
     if command -v mysqld | grep usr.local; then
         cd /usr/local/mysql
     fi
-    mysql_install_db --defaults-file=/var/tmp/my.cnf --force --datadir=/var/lib/mysql
+    mysql_install_db --defaults-file=/var/tmp/my.cnf --force --datadir=${DATADIR:-/var/lib/mysql}
 fi
-echo "Starting mysqld --skip-networking --socket=${SOCKET}"
-mysqld --defaults-file=/var/tmp/my.cnf --user=root --socket=$SOCKET --innodb_log_file_size=48M --skip-networking --datadir=/var/lib/mysql --server-id=0 --skip-log-bin &
+# TODO: Why was this changed to MYSQL_UNIX_PORT, review
```

Comment:
Comment above shows it. See https://dev.mysql.com/doc/refman/8.4/en/environment-variables.html
```suggestion
```